### ISSUE TITLE
Refinement on the message for CS0103, or a new rule.

### ISSUE THREAD

**Issue by otac0n** [dotnet/roslyn#70891]
### Analyzer

**Diagnostic ID**: [CS0103](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1716)

### Describe the improvement

 - Given a CS0103 where the syntax `=>` is available (e.g. replace x = x.foo with x => x.foo)
     (for one case but maybe more, with wider closure exceptions)
       - If the assignment is from an object reference's closure to the object reference (any expression on x that assigns to x but does not close over additional variables)
         (Consider that the assignment may be a typo for a lambda.)
         - Evaluate the replaced lambda (syntactically and async semantically?).
         - If replaced expression is valid (e.g. replace x = x.foo with x => x.foo)
           - Augment the CS0103 with a new warning to check for a typo of '=' vs '=>'

### Describe suggestions on how to achieve the rule

As above, replace the syntax tree.  If the parse is better (according to constrained analysis), update the existing diagnostic.

### Additional context

`.Select(x => x.word)`

vs

`.Select(x = x.word)`

<img width="408" alt="image" src="https://github.com/dotnet/roslyn-analyzers/assets/193584/fc469f35-a07c-479b-9765-c8d1098466a5">

**Comment by huoyaoyuan** [comment]
`x = x.word` is a perfectly valid expression to be passed as an argument if `x` is an identifier.

**Comment by otac0n** [comment]
Yes, I'm aware.  But in the case that this is a *raised* semantic error, I would prefer the message to be improved.
